integration/tests/basic/erc/test_EIP1559.py

```diff
@@ -0,0 +1,1317 @@
+            max_fee_per_gas=max_fee_per_gas,
+            data=contract_iface["bin"],
+            access_list=None,
+        )
+
+        with pytest.raises(expected_exception=expected_exception, match=exception_message_regex):
+            self.web3_client.send_transaction(
+                account=account,
+                transaction=tx_params,
+                timeout=TX_TIMEOUT,
+            )
+
+    def test_insufficient_funds(
+            self,
+    ):
+        sender = self.accounts[0]
+        balance = self.web3_client.get_balance(sender.address)
+        recipient = self.web3_client.create_account()
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id="auto",
+            from_=sender.address,
+            to=recipient.address,
+            value=balance,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas="auto",
+            max_fee_per_gas="auto",
+            data=None,
+            access_list=None,
+        )
+
+        error_msg_regex = r"{'code': -32000, 'message': '.+'}"
+        with pytest.raises(expected_exception=ValueError, match=error_msg_regex):
+            self.web3_client.send_transaction(account=sender, transaction=tx_params)
+
+    def test_too_low_fee(
+            self,
+    ):
+        sender = self.accounts[0]
+        recipient = self.web3_client.create_account()
+
+        base_fee_per_gas = self.web3_client.base_fee_per_gas()
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id="auto",
+            from_=sender.address,
+            to=recipient.address,
+            value=1000,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas=0,
+            max_fee_per_gas=base_fee_per_gas - 1,
+            data=None,
+            access_list=None,
+        )
+
+        error_msg_regex = rf".+ not in the chain after {TX_TIMEOUT} seconds"
+        with pytest.raises(expected_exception=TimeExhausted, match=error_msg_regex):
+            self.web3_client.send_transaction(account=sender, transaction=tx_params, timeout=TX_TIMEOUT)
+
+
+@allure.feature("EIP Verifications")
+@allure.story("EIP-1559: Verify JSON-RPC method eth_maxPriorityFeePerGas")
+@pytest.mark.usefixtures("eip1559_setup")
+class TestRpcMaxPriorityFeePerGas:
+
+    @pytest.mark.need_eip1559_blocks(10)
+    def test_positive(
+            self,
+            json_rpc_client: JsonRPCSession,
+            web3_client: NeonChainWeb3Client,
+    ):
+
+        response = json_rpc_client.send_rpc(method="eth_maxPriorityFeePerGas")
+        assert "error" not in response, response["error"]
+        max_fee_per_gas = int(response["result"], 16)
+
+        fee_history: web3.types.FeeHistory = web3_client._web3.eth.fee_history(10, "pending", [5])
+        estimated_max_fee_per_gas = _fee_history_priority_fee_estimate(fee_history=fee_history)
+        assert abs(max_fee_per_gas - estimated_max_fee_per_gas) <= 2000000000
```

Comment:
why 2000000000?

Reply:
it looks like there's no way to validate eth_maxPriorityFeePerGas precisely, so that the test would be valid on both Neon and Geth

_fee_history_priority_fee_estimate returns a approximate evaluation based on eth_feeHistory

from what I observed, abs(max_fee_per_gas - estimated_max_fee_per_gas) is usually around 1000000000 so I doubled it to avoid test flakiness

if we want to be more precise - the test will be a `neon_only`
how do you think we should proceed?